code/modules/food_and_drinks/kitchen_machinery/processor.dm

```diff
@@ -43,125 +41,44 @@
-	output = /obj/item/reagent_containers/food/snacks/fries
-
-/datum/food_processor_process/sweetpotato
-	input = /obj/item/reagent_containers/food/snacks/grown/potato/sweet
-	output = /obj/item/reagent_containers/food/snacks/yakiimo
-
-/datum/food_processor_process/potato
-	input = /obj/item/reagent_containers/food/snacks/grown/potato
-	output = /obj/item/reagent_containers/food/snacks/tatortot
-
-/datum/food_processor_process/carrot
-	input = /obj/item/reagent_containers/food/snacks/grown/carrot
-	output = /obj/item/reagent_containers/food/snacks/carrotfries
-
-/datum/food_processor_process/soybeans
-	input = /obj/item/reagent_containers/food/snacks/grown/soybeans
-	output = /obj/item/reagent_containers/food/snacks/soydope
-
-/datum/food_processor_process/spaghetti
-	input = /obj/item/reagent_containers/food/snacks/doughslice
-	output = /obj/item/reagent_containers/food/snacks/spaghetti
-
-/datum/food_processor_process/corn
-	input = /obj/item/reagent_containers/food/snacks/grown/corn
-	output = /obj/item/reagent_containers/food/snacks/tortilla
-
-/datum/food_processor_process/parsnip
-	input = /obj/item/reagent_containers/food/snacks/grown/parsnip
-	output = /obj/item/reagent_containers/food/snacks/roastparsnip
-
-/* mobs */
-/datum/food_processor_process/mob/process_food(loc, what, processor)
-	..()
+	picked_slime.forceMove(src)
+
+/obj/machinery/processor/proc/process_food(datum/food_processor_process/recipe, atom/movable/what)
+	if (recipe.output && loc && !QDELETED(src))
+		for(var/i = 0, i < rating_amount, i++)
+			new recipe.output(drop_location())
+	if (ismob(what))
+		var/mob/themob = what
+		themob.gib(TRUE,TRUE,TRUE)
+	else
+		qdel(what)
 
 
-/datum/food_processor_process/mob/slime/process_food(loc, what, obj/machinery/processor/processor)
+/obj/machinery/processor/slime/process_food(datum/food_processor_process/recipe, atom/movable/what)
 	var/mob/living/simple_animal/slime/S = what
-	var/C = S.cores
-	if(S.stat != DEAD)
-		S.loc = loc
-		S.visible_message("<span class='notice'>[C] crawls free of the processor!</span>")
-		return
-	for(var/i in 1 to (C+processor.rating_amount-1))
-		new S.coretype(loc)
-		SSblackbox.add_details("slime_core_harvested","[replacetext(S.colour," ","_")]")
-	..()
-
-/datum/food_processor_process/mob/slime/input = /mob/living/simple_animal/slime
-/datum/food_processor_process/mob/slime/output = null
-
-/datum/food_processor_process/mob/monkey/process_food(loc, what, processor)
-	var/mob/living/carbon/monkey/O = what
-	if (O.client) //grief-proof
-		O.loc = loc
-		O.visible_message("<span class='notice'>Suddenly [O] jumps out from the processor!</span>", \
-				"<span class='notice'>You jump out from the processor!</span>", \
-				"<span class='italics'>You hear chimpering.</span>")
-		return
-	var/obj/bucket = new /obj/item/reagent_containers/glass/bucket(loc)
-
-	var/datum/reagent/blood/B = new()
-	B.holder = bucket
-	B.volume = 70
-	//set reagent data
```

Comment:
Yeah I was told to keep this when I kinda refactored it. Sorry for not doing a better job.